### PR TITLE
Centralized environment variables and their default values in `modules/config`

### PR DESCRIPTION
**Pull request by jsopn**
centralized environment variables in `modules/config` as discussed with @dumbmoron in #463 pull request

### CODE REVIEW

File: src/modules/config.js [hunk 1]
```diff
@@ -26,4 +51,7 @@ export const
     supportedAudio = config.supportedAudio,
     celebrations = config.celebrations,
     links = config.links,
-    sponsors = config.sponsors
+    sponsors = config.sponsors,
+    mode = (process.env.API_URL && !process.env.WEB_URL) ? 'API' :
+        (process.env.WEB_URL && process.env.API_URL) ? 'WEB' : undefined,
```
Comment (dumbmoron) [inline src/modules/config.js:56]: you could use `apiURL` and `webURL` from above here

File: src/modules/stream/types.js [hunk 2]
```diff
@@ -44,8 +44,8 @@ function pipe(from, to, done) {
 }
 
 function getCommand(args) {
-    if (process.env.PROCESSING_PRIORITY && process.platform !== "win32") {
-        return ['nice', ['-n', process.env.PROCESSING_PRIORITY, ffmpeg, ...args]]
+    if (env.processingPriority != '' && process.platform !== "win32") {
```
Comment (dumbmoron) [inline src/modules/stream/types.js:47]:
```suggestion
    if (env.processingPriority && process.platform !== "win32") {
```

File: src/modules/config.js [hunk 3]
```diff
@@ -12,6 +12,31 @@ Object.values(servicesConfigJson.config).forEach(service => {
...
+        apiURL
+    },
+
+    // API mode related environment variables
+    apiEnvs = {
+        apiPort: process.env.API_PORT || 9000,
+        apiName: process.env.API_NAME || 'unknown',
+        corsWildcard: process.env.CORS_WILDCARD !== '0',
+        corsURL: process.env.CORS_URL || '',
+        cookiePath: process.env.COOKIE_PATH || '',
```
Comment (dumbmoron) [inline src/modules/config.js:34]: i'd just let these default to `undefined` if not set (meaning remove the `|| ''`)

Comment (dumbmoron) [reply src/modules/config.js:34]: same with plausibleHostname

File: src/core/api.js [hunk 4]
```diff
@@ -14,8 +14,8 @@ import { generateHmac } from "../modules/sub/crypto.js";
 import { verifyStream, getInternalStream } from "../modules/stream/manage.js";
 
 export function runAPI(express, app, gitCommit, gitBranch, __dirname) {
-    const corsConfig = process.env.CORS_WILDCARD === '0' ? {
-        origin: process.env.CORS_URL,
+    const corsConfig = env.corsWildcard ? {
```
Comment (dumbmoron) [inline src/core/api.js:17]: the meaning of `corsWildcard` is flipped around here

**Review by dumbmoron — CHANGES_REQUESTED** [review]

File: src/modules/stream/types.js (hunk 2 above)
Comment (jsopn) [reply src/modules/stream/types.js:47]: I did it to allow the user to specify number 0 in this variable, do we still remove it?

File: src/modules/config.js (hunk 1 above)
Comment (jsopn) [reply src/modules/config.js:56]: yep, forgot about them, will do

File: src/modules/stream/types.js (hunk 2 above)
Comment (dumbmoron) [reply src/modules/stream/types.js:47]: in that case, i would probably check that `!isNaN(...)`

Comment (jsopn) [reply src/modules/stream/types.js:47]: sure, one moment

**Review by wukko — APPROVED** [review]

File: src/modules/pageRender/elements.js
[the remark this answers is not in the document]
Comment (dumbmoron) [reply src/modules/pageRender/elements.js:1]: 😭

**Comment by dumbmoron** [comment]
thank you for the contribution!!

**Comment by wukko** [comment]
thank you :D

File: src/modules/pageRender/elements.js
[the remark this answers is not in the document]
Comment (jsopn) [reply src/modules/pageRender/elements.js:1]: nice domain! hahaha

**Pull request merged**